code/datums/ability.dm

```diff
@@ -299,6 +315,26 @@
 	if(always_bind)
 		quickbind()
 
+
+/**
+ * checks if the target is within range and the ability is triggerable
+ */
+/datum/ability/proc/target_check(mob/user, atom/target)
+	if(range)
+		if(get_dist(get_turf(user),get_turf(target)) <= range)
+		else return FALSE
```

Comment:
don't do super compact syntax, put the return on its own line, it's more readable that way